### PR TITLE
[ bug #1856 ] ODT Templates - Templating keys are not always recognized

### PR DESCRIPTION
**Pull request by marcosgdf**
Moved changes to own classes DolibarrOdf and DolibarrSegment to not alter the external library.

Now we can handle detection of templating keys with mixed styles

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -28,6 +28,7 @@ English Dolibarr ChangeLog
 - Fix: [ bug #1825 ] External agenda: hide/show checkbox doesn't work
 - Fix: [ bug #1790 ] Email form behaves in an unexpected way when pressing Enter key
 - Fix: Bad SEPA xml file creation
+- Fix: [ bug #1856 ] ODT Templates - Templating keys are not always recognized
```
Comment (rdoursenaud) [inline ChangeLog:31]: You're not supposed to update the ChangeLog anymore. Please see: https://github.com/Dolibarr/dolibarr/blob/develop/CONTRIBUTING.md#general-rules

Comment (marcosgdf) [inline ChangeLog:31]: Only for develop branch. Also I closed the PR because of not working right

Comment (rdoursenaud) [inline ChangeLog:31]: Oh snap, didn't know (remember?) it only applied to the develop branch… Thanks for the heads-up!

**Comment by eldy** [comment]
Even it fix a bug, because PR bring some change into architecture (refactoring by renaming classes), I need to have this request into develop branch to integrate it.

**Pull request closed without merging**